### PR TITLE
[apps/ca] Clean and simplify CA-certificate output path/filename construction

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
##### Checklist
- [X] CLA is signed

##### Description of change
Follows discussions in PR #1569
( Found a subtle bug while rereading my changes. => was merged as 0db1fb3fc13c4b1a2b916efbb374f40579b1398f )
Applicable to master and 1.1.0 only.

### CODE REVIEW

**Comment by FdaSilvaYY** [comment]
@richsalz : Yes, i forget to drop this one; I use it as I'm bored of CI mail notifications.
BTW, someone ask a few days ago about this:
https://docs.travis-ci.com/user/customizing-the-build/#Skipping-a-build

File: apps/ca.c [hunk 1]
```diff
@@ -985,36 +979,34 @@ int ca_main(int argc, char **argv)
...
-            BIO *Cout = NULL;
-            ASN1_INTEGER *serialNumber = X509_get_serialNumber(x);
-            int k;
-            char *n;
-
-            x = sk_X509_value(cert_sk, i);
 
-            j = ASN1_STRING_length(serialNumber);
-            p = (const char *)ASN1_STRING_get0_data(serialNumber);
+        new_cert[PATH_MAX - 1] = '\0';
```
Comment (levitte) [inline apps/ca.c]: There's really no need for this at all, since the following `strcpy` will place the NUL character in its correct position anyway.

**Comment by FdaSilvaYY** [comment]
Ping ! old PR that fix some remarks raised in #1569.
It should improve the coverage

**Comment by FdaSilvaYY** [comment]
Thanks !!

**Comment by FdaSilvaYY** [comment]
Ping !
After two Approvals, it's ready for push, no ? ;)

**Comment by richsalz** [comment]
Squashed and merged.  Thank you for the cleanups!

**Pull request closed without merging**